cumulusci/conftest.py

```diff
@@ -75,3 +78,19 @@ def fallback_orgconfig():
 vcr = fixture(salesforce_vcr, scope="module")
 
 create_task_fixture = fixture(create_task_fixture, scope="function")
+
+
+# TODO: This should also chdir to a temp directory which
+#       can represent the repo-root, but that will require
+#       test case changes.
+@pytest.fixture(autouse=True, scope="session")
+def patch_home(request):
+    with TemporaryDirectory(prefix="fake_home_") as home, mock.patch(
+        "pathlib.Path.home", lambda: Path(home)
+    ), mock.patch.dict(
+        os.environ,
+        {"HOME": home, "USERPROFILE": home, "CUMULUSCI_KEY": "0123456789ABCDEF"},
```

Comment:
I like isolating the test environment, but:
- Call this `patch_home_and_env` to make it more obvious that this is part of what it does
- What's the argument for including CUMULUSCI_KEY in the default environment? Running cci with it set is relatively uncommon.

Reply:
It was necessary because CCI doesn't run without a key so you either supply one or use one from the user's environment. I got error messages before I added this. Of course setting this environment variable is not the only way to fake a key, but it was one easy way.

Reply:
@prescod I guess this is fine, but there are a couple of tests which do `del os.environ["CUMULUSCI_KEY"]` to make sure we get coverage of the code which uses keyring to generate a key when one is missing. We should update them to make sure they only remove it temporarily.

Reply:
I've ensured they use a nested patch now.